src/libstore/binary-cache-store.cc

```diff
@@ -142,17 +142,10 @@ struct FileSource : FdSource
     }
 };
 
-void BinaryCacheStore::addToStore(const ValidPathInfo & info, Source & narSource,
-    RepairFlag repair, CheckSigsFlag checkSigs)
+ref<const ValidPathInfo> BinaryCacheStore::addToStoreCommon(
+    Source & narSource, RepairFlag repair, CheckSigsFlag checkSigs,
+    std::function<ValidPathInfo(HashResult)> mkInfo)
 {
-    assert(info.narSize);
-
-    if (!repair && isValidPath(info.path)) {
-        // FIXME: copyNAR -> null sink
-        narSource.drain();
```

Comment:
Are we sure we don't need this anymore? Previously the drain was necessary to ensure the postcondition that the NAR is completely read, to avoid leaving the connection in an unknown state. `FramedSource` should address that, but maybe there are still places where it could happen.

Reply:
Oh I see it's still drained elsewhere. Sorry :-)